src/viam/errors.py

```diff
@@ -93,3 +93,13 @@ class ValidationError(ViamGRPCError):
     def __init__(self, message: str):
         self.message = message
         self.grpc_code = Status.INVALID_ARGUMENT
+
+
+class NoCaptureToStoreError(ViamGRPCError):
+    """
+    Exception raised in filter module to signal that data capture should not be stored
+    """
+
+    def __init__(self):
+        self.message = "no capture from filter module"
```

Comment:
Yep the error handling is partially dependent on the text - added an inline comment. 

I agree that this is brittle, especially if we ever want to change the message (though no plans as of now to need to do so) - am definitely open to ideas on how to keep this in sync across repos.

In terms of the custom proto message: are you suggesting returning a custom Response for this case instead of an error, or storing the error string content in a proto which the SDKs could import to build an error with? If it's the former - that would be pretty involved since we'd need to change every API that's touched by a collector to handle this response; potentially worth doing in the future but probably not for V0 of this feature. Not sure if I know how the latter would work since protos (as far as I've seen) are used to specify the structure of an object and not store any value - let me know if I'm misunderstanding something!

Reply:
No misunderstanding! I was initially thinking the former but I haven't looked to deeply into using enums/statuses to convey the correct error state. But regardless, I do think that it's probably outside the scope of this PR. I do think that if we don't do it now, though, it won't ever be done (just one of those things that will never get prioritized) and it will remain a potential fail point. 

But with your new comment, hopefully no one changes it :)